zerver/management/commands/delete_user.py

```diff
@@ -0,0 +1,43 @@
+from argparse import ArgumentParser
+from typing import Any
+
+from zerver.lib.actions import do_delete_user
+from zerver.lib.management import CommandError, ZulipBaseCommand
+from zerver.models import UserProfile
+
+
+class Command(ZulipBaseCommand):
+    help = (
+        "Delete a user, including all messages sent by them and personal messages received by them."
+    )
+
+    def add_arguments(self, parser: ArgumentParser) -> None:
+        parser.add_argument(
+            "-f",
+            "--for-real",
+            action="store_true",
+            help="Actually delete the user. Default is a dry run.",
+        )
+        parser.add_argument("email", metavar="<email>", help="email of user to deactivate")
+        self.add_realm_args(parser)
+
+    def handle(self, *args: Any, **options: Any) -> None:
+        realm = self.get_realm(options)
+        user_profile = self.get_user(options["email"], realm)
+
+        print(
+            "{} has {} active bots that will be deactivated as a result of the user's deletion.".format(
+                user_profile.delivery_email,
+                UserProfile.objects.filter(
+                    is_bot=True,
+                    is_active=True,
+                    bot_owner=user_profile,
+                ).count(),
```

Comment:
Let's extract a `get_owner_bot_users` function from `do_deactivate_user` and call that from both here and there, so the implementations of the function used here and for actual deactivation cannot get out of sync.